volume/local/local_unix.go

```diff
@@ -136,6 +138,22 @@ func (v *localVolume) mount() error {
 			mountOpts = strings.Replace(mountOpts, "addr="+addrValue, "addr="+ipAddr.String(), 1)
 		}
 	case "cifs":
+		resolveDeviceName := true
+		// Don't perform the DNS resolution if the addr or ip option was passed
+		// explicitly by the user.
+		opts := strings.Split(mountOpts, ",")
+		for _, optKV := range opts {
+			opt, _, ok := strings.Cut(optKV, "=")
+			if ok && (opt == "addr" || opt == "ip") {
+				log.G(context.TODO()).Debugf(`not performing dns resolution of device %q, because "addr" or "ip" opt was passed`, v.opts.MountDevice)
+				resolveDeviceName = false
+				break
+			}
```

Comment:
Because `addr` is an alias for `ip` (both expect an address), I think we should preserve the old behaviour for CIFS by resolving `addr` if it's a hostname.

Then it'll continue to work for people who've been following [our docs](https://docs.docker.com/storage/volumes/#create-cifssamba-volumes), and it won't be necessary for them to [rebuild the volume with 'addr=' removed](https://github.com/moby/moby/pull/46863#issuecomment-1904472670).

That rewrite of the `addr` option before passing it to the kernel is what we were doing in v24, and we're still doing in v25 for NFS, to allow people to avoid hard-coding an address.

Reply:
Yeah, honestly, it was a bit hacky from the start (and we don't have a setup in our CI to verify). I agree though that we should probably preserve existing behavior (if that worked for some?)

Reply:
Restored the old behavior for `addr`.